src/Tickets/Commerce/Gateways/Stripe/Merchant.php

```diff
@@ -173,4 +182,88 @@ public function save_signup_data( array $signup_data ) {
 
 		return update_option( $this->get_signup_data_key(), $signup_data );
 	}
+
+	/**
+	 * Empty the signup data option and void the connection
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function delete_signup_data() {
+		return update_option( $this->get_signup_data_key(), [] );
+	}
+
+	/**
+	 * Validate if this Merchant is allowed to connect to the TEC Provider
+	 *
+	 * @since TBD
+	 *
+	 * @return string 'valid' if the account is permitted, or a string with the notice slug if not
+	 */
+	public function validate_account_is_permitted() {
+		$status = tribe( Settings::class )->connection_status;
+		if ( empty( $status ) ) {
+			tribe( Settings::class )->set_connection_status();
+			$status = tribe( Settings::class )->connection_status;
+		}
+
+		$is_licensed = \TEC\Tickets\Commerce\Settings::is_licensed_plugin();
+
+		if ( $is_licensed ) {
+			return 'valid';
+		}
+
+		if ( ! $this->country_is_permitted( $status ) ) {
+			return 'tc-stripe-country-denied';
+		}
+
+		return 'valid';
+	}
+
+	/**
+	 * Determine if a stripe account is listed in a permitted country
+	 *
+	 * @since TBD
+	 *
+	 * @param array $status the connection status array
+	 *
+	 * @return bool
+	 */
+	public function country_is_permitted( $status ) {
+		// @todo figure out any other exclusions
+		return 'BR' !== $status['country'];
```

Comment:
Shall we put `BR` in an array and do a check against the array in case we need to add more?